### PR TITLE
Remove variable type list(object) and create default values in maintf, implement "disable" for each rule

### PR DESCRIPTION
**Pull request by ivayloivanov388**
Remove variable type list(object) and create default values in maintf, implement "disable" for each rule

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/firewall-rules/README.md /tmp/tmp.ut2xTxdqeQ/generate_docs/workspace/modules/firewall-rules/README.md
45c45
< | rules | List of custom rule definitions (refer to variables file for syntax). | <pre>{<br>    name                    = string<br>    description             = string<br>    direction               = string<br>    priority                = number<br>    ranges                  = list(string)<br>    source_tags             = list(string)<br>    source_service_accounts = list(string)<br>    target_tags             = list(string)<br>    target_service_accounts = list(string)<br>    disabled               = bool<br>    allow = list(object({<br>      protocol = string<br>      ports    = list(string)<br>    }))<br>    deny = list(object({<br>      protocol = string<br>      ports    = list(string)<br>    }))<br>    log_config = object({<br>      metadata = string<br>    })<br>  }</pre> | `[]` | no |
---
> | rules | List of custom rule definitions (refer to variables file for syntax). | `any` | n/a | yes |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

**Comment by ivayloivanov388** [comment]
up

File: modules/firewall-rules/variables.tf [hunk 1]
```diff
@@ -24,27 +24,4 @@ variable "network_name" {
 
 variable "rules" {
   description = "List of custom rule definitions (refer to variables file for syntax)."
-  default     = []
-  type = list(object({
```
Comment (bharathkkb) [inline modules/firewall-rules/variables.tf:28]: Why are we removing type info? IIUC to support disabled we can just append to existing rule object?

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @ivayloivanov388

**Comment by ivayloivanov388** [comment]
bump

**Comment by bharathkkb** [comment]
@ivayloivanov388 Please address https://github.com/terraform-google-modules/terraform-google-network/pull/355#discussion_r846921864 and the failing lint check https://github.com/terraform-google-modules/terraform-google-network/pull/355#issuecomment-1069697212

**Comment by ivayloivanov388** [comment]
@bharathkkb  its done

**Pull request closed without merging**